### PR TITLE
Rework - 431 company roles and features

### LINKED ISSUE #431: Rework: company roles logic
Currently, the companyrep needs to select a role when inviting company members. When a member is a speaker, he can request for a presentation. When a booth is approved, the companyrep can invite a member as a booth owner. Once done, the companyrep cannot manage the booth ownership.

I suggest the following for booth ownership:
- Companyrep can invite and remove members
- When creating a booth registration, he selects one of the members (including himself) as booth owner
- After approval, he can update the booth ownership
- When he deletes the boothowner the ownership is automatically set to the companyrep

And for the presentations:
- Any user can request for a presentation
- When the user has role 'company member', the company is associated with the presentation (optional: the user can select whether or not the presentation should be attached to the company)
- Companyrep can see all the presentations that are associated with the company
- Whe the companyrep removes the member that has a presentation, the presentation will be detached from the company

### PR DESCRIPTION
**Pull request by v-stamenova**
# Description

This branch introduces the new feature of action based roles within the companies. This would mean that once a company representative gain access to send out invitations they no longer need to specify role in advance. This means that once the new employee registers via the company email they'll get the chance to become a speaker (either request or join presentation) or booth owner (either request or join booth).

closes #431 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## What needs to be tested

All tests would require as prerequisite a registration as company rep and approval of the company (use the new function to speed up approval process `sail artisan app:approve-last-created-company`)

- [ ] Invite people (preferably at least 4)
- [ ] Register as one of them and look around and make sure you can sign up as booth owner and as speaker
- [ ] After that request presentation
- [ ] After requesting ensure you cannot reach booth owners permissions
- [ ] Register as one of them and join as co-speaker
- [ ] After joining as co-speaker ensure you cannot reach booth owners permissions
- [ ] Register as a one of them and request a booth
- [ ] After requesting ensure you cannot reach speaker permissions
- [ ] Register as a one of them and join as booth owner
- [ ] After joining as booth owner ensure you cannot reach speaker permissions

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Livewire/Presentation/JoinAsSpeakerModal.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function save()
+    {
+        $this->authorize('joinAsCospeaker', $this->presentation);
+
+        Auth::user()->joinPresentation($this->presentation, 'speaker');
+
+        return redirect(route('presentations.show', $this->presentation))
+            ->with('status', 'Presentation successfully updated.');
+    }
```
Comment (TimKardol) [inline app/Livewire/Presentation/JoinAsSpeakerModal.php:35]: image

Don't know if this function is the cause of the error.
Clicked on the 'join as co-speaker' button, then the error occured.

File: resources/views/components/dashboards/blocks/company-role-decider.blade.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    </div>
+                    @foreach($speakerButtons as $label => $route)
+                        @if ($loop->last && count($speakerButtons) > 1)
+                            <div class="pt-2">
+                                <span>or</span>
+                            </div>
+                        @endif
+                        <div class="pt-2">
+                            <!-- As per current implementation, it's an array it must be a POST request;
+                            refer the backend of the component for more-->
```
Comment (TimKardol) [inline resources/views/components/dashboards/blocks/company-role-decider.blade.php:35]: image

See my other comment.

**Review by TimKardol — CHANGES_REQUESTED** [review]

**Review by TimKardol — APPROVED** [review]

File: resources/views/auth/registration-via-invitation.blade.php [hunk 3]
```diff
@@ -1,63 +1,125 @@
...
+        <div class="grid grid-cols-1 md:grid-cols-7 w-full h-full md:h-[80vh] bg-white dark:bg-gray-800 rounded-md">
+            <div id="pretty-slide"
+                 class="h-full col-span-3 hidden md:block rounded-md">
+                <div class="h-full rounded-md" style="overflow: clip">
+                    <div class="relative h-full">
+                        <img class="h-full object-cover" src="/img/market.jpg" alt="market">
+                        <div class="gradient absolute inset-0"
+                             style="background: linear-gradient(to bottom right, rgba(54, 102, 255, 0.7), rgba(184, 98, 214, 0.7));"></div>
+                        <div class="absolute inset-0 flex justify-center items-center" style="z-index: 3">
+                            <h2 class="text-4xl font-bold text-white drop-shadow-md text-center leading-tight">We are in
```
Comment (dwaard) [inline resources/views/auth/registration-via-invitation.blade.php:15]: All this empty space really necessary?

File: resources/views/auth/registration-via-invitation.blade.php [hunk 4]
```diff
@@ -1,63 +1,125 @@
...
-                <x-label for="password_confirmation" value="{{ __('Confirm Password') }}" class="after:content-['*'] after:text-red-500"/>
-                <x-input id="password_confirmation" class="block mt-1 w-full" type="password"
-                         name="password_confirmation" required autocomplete="new-password"/>
-            </div>
+                                    <div id="password-rules"
+                                         class="hidden text-sm text-gray-700 dark:text-gray-300 mt-2 pl-2">
+                                        <p>Password should be of the following format:</p>
+                                        <ul class="pl-5 pt-0.5">
+                                            <li>
+                                                <p id="length-false" class="before:content-['✗_'] text-red-500">length
```
Comment (dwaard) [inline resources/views/auth/registration-via-invitation.blade.php:67]: Formatting?

File: resources/views/auth/registration-via-invitation.blade.php [hunk 5]
```diff
@@ -1,63 +1,125 @@
...
-
-                            <div class="ml-2">
-                                {!! __('I agree to the :terms_of_service and :privacy_policy', [
-                                        'terms_of_service' => '<a target="_blank" href="'.route('terms.show').'" class="underline text-sm text-gray-600 dark:text-gray-400 hover:text-gray-900 dark:hover:text-gray-100 rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 dark:focus:ring-offset-gray-800">'.__('Terms of Service').'</a>',
-                                        'privacy_policy' => '<a target="_blank" href="'.route('policy.show').'" class="underline text-sm text-gray-600 dark:text-gray-400 hover:text-gray-900 dark:hover:text-gray-100 rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 dark:focus:ring-offset-gray-800">'.__('Privacy Policy').'</a>',
-                                ]) !!}
-                            </div>
+                                                <p id="length-true" class="hidden before:content-['✓_'] text-green-500">
+                                                    length is at
```
Comment (dwaard) [inline resources/views/auth/registration-via-invitation.blade.php:73]: Formatting?

**Review by dwaard — CHANGES_REQUESTED** [review]
I tested as good as I could. These are my observations:

- I was able to invite users, and accept an invitation.
- When inviting, the 'Add' button might be renamed to 'Invite' or 'Send invitation'
- After sending, there is no confirmation message that the invitation is sent
- As a user who accepted an invitation, I could look around, add myself as a co-speaker, but cannot request a booth
- As any user I cannot see who the speaker(s) of a presentation are
- Another invited user was able to request a booth
- As an admin: ERROR `View [crew.details.booth] not found.`
- As an invited user, I am not able to request a presentation. It might be because there was already a presentation associated with the company
- As company rep, I do not have clear overview of which presentations and booths are associated to my company

Not sure what in this list is correct or not and if the error is part of this PR

**Pull request merged**